lib/iris/fileformats/rules.py

```diff
@@ -1034,6 +1030,15 @@ def loadcubes_user_callback_wrapper(cube, field, filename):
         cube = iris.io.run_callback(user_callback, cube, field, filename)
```

Comment:
@dkillick For me, this is the nub of the problem, and it's not your doing.

I believe `lines +1034-1035` should now simply be the following:
```python
result = cube
if user_callback is not None:
    result = user_callback(cube, field, filename)
return result
```

I believe the intent of the `loadcubes_user_callback_wrapper` was simply to 

1. raise the deprecation warning
1. fire any registered custom user rules, then
1. execute the `user_callback`. 

The `user_callback` should **not** be executed here by `iris.io.run_callback`. That's the problem.

Rather, it is the `loadcubes_user_callback_wrapper` that should be executed by `iris.io.run_callback` downstream within `_load_pairs_from_fields_and_filenames`, as was prior to your changes.

So make the above change, then back-out your patch to [_load_pairs_from_fields_and_filenames](https://github.com/SciTools/iris/pull/2059/files#diff-71a6ac4e2a835fb3910a8a8a8045474eR952) and replace it with
```python
cube = iris.io.run_callback(user_callback, cube, field, filename)
if cube is None:
    continue
```